bson/_cbsonmodule.c

```diff
@@ -2342,7 +2342,7 @@ static int _element_to_dict(PyObject* self, const char* string,
     if (name_length > BSON_MAX_SIZE || position + name_length >= max) {
         PyObject* InvalidBSON = _error("InvalidBSON");
         if (InvalidBSON) {
-            PyErr_SetNone(InvalidBSON);
+            PyErr_SetString(InvalidBSON, "invalid field name");
```

Comment:
Say why it's invalid?

Reply:
Done. I went with "field name too large". This can happen when the BSON itself gets corrupted or mangled. For example the null terminator was overwritten or the field was internally truncated like in the test case below.